hazelcast/src/main/java/com/hazelcast/internal/metrics/impl/DoubleGaugeImpl.java

```diff
@@ -84,8 +84,8 @@ private static double getMetricValue(String gaugeName, Object source, ProbeFunct
                 DoubleProbeFunction doubleFunction = (DoubleProbeFunction) function;
                 return doubleFunction.get(source);
             }
-        } catch (Exception e) {
-            metricsRegistry.logger.warning("Failed to access the probe: " + gaugeName, e);
+        } catch (Throwable t) {
+            metricsRegistry.logger.warning("Failed to access the probe: " + gaugeName, t);
```

Comment:
To prevent goblin up throwables, I believe we should only log the exception and do a sneaky throw to rethrow the throwable.

Exception handling in Java sucks.

Reply:
> To prevent goblin up throwables, I believe we should only log the exception and do a sneaky throw to rethrow everything than an exception.
> 
> Exception handling in Java sucks.

I did try this, but it made the actual invocation slower. I don't have the numbers, but it was noticeable.

I don't *think* the problem is catching exceptions and rethrowing (obviously that's slow), but wrapping the statement in a `try` also has other side effects regarding encapsulation that added overhead.

Is there a specific reason your not liking it?

Reply:
It isn't a matter of me not liking it, we have some weird exception handlers in the system that rely on certain throwables to be properly propagated.

Reply:
943c4940847cf28642360ae5407fbe0026e3a704